src/ja/nicomanga/src/eu/kanade/tachiyomi/extension/ja/nicomanga/Nicomanga.kt

```diff
@@ -0,0 +1,165 @@
+package eu.kanade.tachiyomi.extension.ja.nicomanga
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.Jsoup
+import java.net.URL
+
+class Nicomanga : HttpSource() {
+    companion object {
+        val thumbnailURLRegex: Regex = "background-image:[^;]url\\s*\\(\\s*'([^?']+)".toRegex()
+
+        val statusRegex: Regex = "(?<=-)[^.]+".toRegex()
+
+        val urlRegex: Regex = "(?<=manga-)[^/]+(?=\\.html\$)".toRegex()
+
+        val floatRegex: Regex = "\\d+(?:\\.\\d+)?".toRegex()
+
+        val chapterIdRegex: Regex = "(?<=imgsListchap\\()\\d+".toRegex()
+    }
+
+    override val baseUrl: String = "https://nicomanga.com"
+
+    override val lang: String = "ja"
+
+    override val name: String = "Nicomanga"
+
+    override val supportsLatest: Boolean = true
+
+    override val client: OkHttpClient = network.cloudflareClient
+
+    private fun mangaListParse(response: Response): MangasPage {
+        val doc = Jsoup.parse(response.body.string())
+        val hasNextPage =
+            if (doc.select(".pagination li:last-of-type").size > 0 &&
+                doc.select(".pagination li:last-of-type")[0].text() == "»"
+            ) {
+                doc.select(".pagination li:last-of-type a.disabled").size == 0
+            } else {
+                doc.select(".pagination li:last-of-type a.active").size == 0
+            }
+        val mangas = doc.select(".row > .thumb-item-flow").map { manga ->
+            SManga.create().apply {
+                val relURL = manga.selectFirst(".series-title a")?.attr("href") ?: ""
+                setUrlWithoutDomain(URL(URL(baseUrl), relURL).toString())
```

Comment:
```suggestion
                setUrlWithoutDomain(manga.selectFirst(".series-title a")!!.absUrl("href"))
```
Don't use `?` when a non-empty value is required.